src/classes/CW_AudienceCSVProcessHelper.cls

```diff
@@ -0,0 +1,406 @@
+   	}
+   	update CsvStageList;
+   }
+    
+    
+   public void processImportDataToCamp(List<CW_CSV_Stage_AudienceMembers__c> scope){
+   	//We have set groups , Members adn freeBets. So we can impotr all with seted reference others records a ignored.
+   	CsvStageList=scope;
+   	//firts step crate non exited camping rewards and load existings
+   	loadCreateCampaignReward();
+   	
+   	//Next step is  CreateAudiece sschedule
+   	createLoadAudienceGroupSchdule();
+   	
+   	createAudinceGroupScheduleReward();
+   	
+   	//And olmost FInala Step Is  
+   	createAudinceMember();
+   	//And Final step
+ 		createAudinceMemberReward();
+   } 
+    
+    
+   //Private methods for step 2 
+   private map<string,Id> CreateSheildAccounts(Set<String> ContactNameList){
+
+  // Rigth now crate Scheiled Account.
+    	List<Account> newContacs= new List<Account>();
+      if(ContactNameList.Size()>0 ){
+      	Id personAccountRTId = [select id from RecordType where SobjectType = 'Account' AND IsPersonType = true limit 1].id;
+         for(String playerId:ContactNameList ){
+         	Account acc = new Account();
+            acc.FirstName = playerId;
+            acc.LastName = playerId;
+            acc.Account_Name__pc = playerId;
+            acc.RecordtypeId = TestDataUtil.personAccountRTId;
+            acc.Requires_Update__pc = true;
+            newContacs.Add(acc);
+         }
+        	Insert newContacs; 
+      }
+      map<string,Id> reslt= new map<string,Id>();
+      for(Contact rec: [SELECT id, Account_Name__c FROM Contact WHERE Account_Name__c=:ContactNameList]){
+      	reslt.put(rec.Account_Name__c.ToUpperCase(),rec.id);
+      }
+      return reslt;
+   }
+   
+    private map<string,Id> CreateAudienceGroups(Set<String> GroupNameList ){
+   	list<CW_Audience_Group__c> grpList= new list<CW_Audience_Group__c>();
+      for(String grpName:GroupNameList ){
+     		CW_Audience_Group__c grp = new CW_Audience_Group__c(Campaign__c = CampaignID);
+      	grp.Name=grpName;
+         grp.Is_Control_Group__c = false;
+         grp.Description__c = grpName;
+         grpList.Add(grp);
+      }
+      Insert grpList;
+      map<string,Id> reslt= new map<string,Id>();
+      for(CW_Audience_Group__c rec:grpList ){
+      	reslt.put(rec.Name.ToUpperCase(),rec.id);
+      }
+		return reslt;
+   } 
+   
+   
+   //End OF step 2 - Step 3
+   
+   //Create Campiang Rewards - WORKING OK
+   //Add Error chcking on Insert Records
+   private void loadCreateCampaignReward(){
+   	campRewardMap= new Map<ID,CW_Campaign_Reward__c>();
+   	Map<String,CW_Campaign_Reward__c> campFreeBetdMap= new Map<String,CW_Campaign_Reward__c>();
+   	for(CW_CSV_Stage_AudienceMembers__c rec:CsvStageList ){
+   		campFreeBetdMap.Put(rec.FreeBetId__c.ToUpperCase(),null);
+   	}
+   	for(CW_Campaign_Reward__c rec: [SELECT Id,Reward_Library__c, Reward_Library__r.FreeBet_Id__c,Reward_Library__r.Amount__c,Reward_Library__r.Currency__c FROM CW_Campaign_Reward__c
```

Comment:
Problem: Unbound SOQL query. Reason: a SOQL query can return up to 50,000 records. If this number is exceeded a runtime exception is thrown. Suggestion: cap the max number of records returned adding a LIMIT clause to the query.